kong/plugins/ip-restriction/handler.lua

```diff
@@ -52,31 +53,67 @@ local function match_bin(list, binary_remote_addr)
 end
 
 
-function IpRestrictionHandler:access(conf)
+local function do_exit(status, message, is_http)
```

Comment:
I think we can remove this arg `is_http`.

In `stream` subsystem we will call `preread()`, and in `http` subsystem we will call `access(()`.

So we could do this to eliminate the runtime cost:
```
local is_http_subsystem = ngx.config.subsystem == "http"
local do_exit
if is_http_subsystem then
  do_exit = function() ... end
else
  do_exit = function() ... end
end
```